src/tests/ftest/io/eight_servers_mpiio.py

```diff
@@ -49,7 +49,9 @@ def setUp(self):
 
     def tearDown(self):
         try:
-            if self.pool is not None and self.pool.attached:
+            if self.hostfile_clients is not None:
+                os.remove(self.hostfile_clients)
+            if self.pool is not None:
```

Comment:
This needs to go into another `try`/`finally` block to ensure that the `tearDown()` below happens even if `self.pool.destroy()` throws an exception.

Reply:
Done